context-propagation/src/main/java/io/smallrye/opentracing/SmallRyeOpenTracingContextProvider.java

```diff
@@ -0,0 +1,43 @@
+package io.smallrye.opentracing;
+
+import java.util.Map;
+
+import org.eclipse.microprofile.context.spi.ThreadContextProvider;
+import org.eclipse.microprofile.context.spi.ThreadContextSnapshot;
+
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+
+public class SmallRyeOpenTracingContextProvider implements ThreadContextProvider {
+
+    @Override
+    public ThreadContextSnapshot currentContext(Map<String, String> props) {
+        Tracer activeTracer = GlobalTracer.get();
+        Span activeSpan = activeTracer.activeSpan();
+
+        return () -> {
+            if (activeSpan != null) {
+                Scope s = activeTracer.scopeManager().activate(activeSpan, false);
+                return () -> {
+                    s.close();
+                };
+            }
+
+            return () -> {
+            };
+        };
+    }
+
+    @Override
+    public ThreadContextSnapshot clearedContext(Map<String, String> props) {
```

Comment:
I am not familiar with this API. Could you please comment why we don't have to handle OT context here?

Reply:
I tried implementing removing the active span from the propagated context

Reply:
`clearedContext` is for when user specifically wants given context *not* to propagate.
Now, I might be wrong as I don't know much about opentracing, but it looks like you are propagating it nonetheless?